core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java

```diff
@@ -734,114 +757,85 @@ public String getDestinationName() {
 	}
-	public String getDestinationType() {
+	public void setDestinationTypeEnum(DestinationTypeEnum destinationType) {
+		this.destinationType=destinationType;
+	}
+	public DestinationTypeEnum getDestinationTypeEnum() {
 		return destinationType;
 	}
 
+	public boolean isUseTopicFunctions() {
+		return useTopicFunctions;
+	}
+
 	/**
 	 * Sets the JMS-acknowledge mode. This controls for non transacted listeners the way messages are acknowledged.
 	 * See the jms-documentation.
 	 */
+	@Deprecated
+	@ConfigurationWarning("please use attribute acknowledgeMode instead")
 	public void setAckMode(int ackMode) {
-		this.ackMode = ackMode;
+		for (AcknowledgeModeEnum ackModeEnum : AcknowledgeModeEnum.values()) {
+			if (ackModeEnum.acknowledgeMode == ackMode) {
+				this.ackMode = ackModeEnum;
+				return;
+			}
+		}
+		this.ackMode = AcknowledgeModeEnum.NOT_SET;
 	}
-	public int getAckMode() {
+
+	public AcknowledgeModeEnum getAckModeEnum() {
 		return ackMode;
 	}
 
-	/**
-	 * Convencience function to convert the numeric value of an (@link #setAckMode(int) acknowledgeMode} to a human-readable string.
-	 */
-	public static String getAcknowledgeModeAsString(int ackMode) {
-		String ackString;
-		if (Session.AUTO_ACKNOWLEDGE == ackMode) {
-			ackString = "Auto";
-		} else
-			if (Session.CLIENT_ACKNOWLEDGE == ackMode) {
-				ackString = "Client";
-			} else
-				if (Session.DUPS_OK_ACKNOWLEDGE == ackMode) {
-					ackString = "Dups";
-				} else {
-					ackString = "none";
-				}
-
-		return ackString;
-	}
 
 	/**
 	 * String-version of {@link #setAckMode(int)}
 	 */
-	@IbisDoc({"", "auto_acknowledge"})
+	@IbisDoc({"3", "Acknowledge mode, can be one of ('auto' or 'auto_acknowledge'), ('dups' or 'dups_ok_acknowledge') or ('client' or 'client_acknowledge')", "auto_acknowledge",})
 	public void setAcknowledgeMode(String acknowledgeMode) {
-
-		if (acknowledgeMode.equalsIgnoreCase("auto") || acknowledgeMode.equalsIgnoreCase("AUTO_ACKNOWLEDGE")) {
-			ackMode = Session.AUTO_ACKNOWLEDGE;
-		} else
-			if (acknowledgeMode.equalsIgnoreCase("dups") || acknowledgeMode.equalsIgnoreCase("DUPS_OK_ACKNOWLEDGE")) {
-				ackMode = Session.DUPS_OK_ACKNOWLEDGE;
-			} else
-				if (acknowledgeMode.equalsIgnoreCase("client") || acknowledgeMode.equalsIgnoreCase("CLIENT_ACKNOWLEDGE")) {
-					ackMode = Session.CLIENT_ACKNOWLEDGE;
-				} else {
-					// ignore all ack modes, to test no acking
-					log.warn("["+getName()+"] invalid acknowledgemode:[" + acknowledgeMode + "] setting no acknowledge");
-					ackMode = -1;
-				}
-
-	}
-	/**
-	 * String-version of {@link #getAckMode()}
-	 */
-	public String getAcknowledgeMode() {
-		return getAcknowledgeModeAsString(getAckMode());
+		for (AcknowledgeModeEnum ackModeEnum : AcknowledgeModeEnum.values()) {
+			if (ackModeEnum.shortName.equalsIgnoreCase(acknowledgeMode) || ackModeEnum.name().equalsIgnoreCase(acknowledgeMode)) {
+				this.ackMode = ackModeEnum;
+				return;
+			}
+		}
```

Comment:
Ook hier zou je een parse methode voor kunnen gebruiken. Wanneer hij niet kan parsen kan je een illegalArgumentExeption gooien. 

Ik stel voor hier een (tijdelijke) try catch om te zetten en een ConfigurationWarning te geven wanneer een invalide waarde op is gegeven. Vroeger kon je van alles verzinnen om AckMode uit te zetten, nu is er een speciale waarde die men kan gebruiken.